packages/webpack/src/plugins/webpack-nx-build-coordination-plugin.ts

```diff
@@ -56,44 +45,55 @@ export class WebpackNxBuildCoordinationPlugin {
     this.currentlyRunning = 'nx-build';
     return new Promise<void>((res) => {
       try {
-        const cp = exec(this.buildCmd);
+        this.buildCmdProcess = exec(this.buildCmd);
 
-        cp.stdout.pipe(process.stdout);
-        cp.stderr.pipe(process.stderr);
-        cp.on('exit', () => {
+        this.buildCmdProcess.stdout.pipe(process.stdout);
+        this.buildCmdProcess.stderr.pipe(process.stderr);
+        this.buildCmdProcess.on('exit', () => {
           res();
         });
-        cp.on('error', () => {
+        this.buildCmdProcess.on('error', () => {
           res();
         });
         // eslint-disable-next-line no-empty
       } catch (e) {
         res();
-      } finally {
-        this.currentlyRunning = 'none';
       }
+    }).then(() => {
+      this.currentlyRunning = 'none';
+      this.buildCmdProcess = null;
     });
   }
+
+  private createFileWatcher() {
+    const runner = new BatchFunctionRunner(() => this.buildChangedProjects());
+    return daemonClient.registerFileWatcher(
+      {
+        watchProjects: 'all',
+      },
+      (err, { changedProjects, changedFiles }) => {
+        if (err === 'closed') {
+          output.error({
+            title: 'Watch connection closed',
+            bodyLines: [
+              'The daemon has closed the connection to this watch process.',
+              'Please restart your watch command.',
+            ],
+          });
+          process.exit(1);
+        }
+
+        if (this.buildCmdProcess) {
+          this.buildCmdProcess.kill(2);
+          this.buildCmdProcess = null;
+        }
```

Comment:
If new changes are detected, kill the current process running the build allowing the Promise to resolve, allowing the next item in the queue to be processed.

Consideration: An API for the `BatchFunctionRunner` to clear the queue before queuing a new build. There could be a potential issue where we have 3 builds queued, we detect changes, and we only cancel the first build, but there are still builds queued. 
However, I haven't been able to produce this scenario in testing.